vcr/holder/oidc4vci_wallet.go

```diff
@@ -170,6 +184,57 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
 	return nil
 }
 
+func supportedCredentialFormat(offeredCredential map[string]interface{}) bool {
+	switch offeredCredential["format"] {
+	case oidc4vci.VerifiableCredentialJSONLDFormat:
+		return true
+	default:
+		return false
+	}
+}
+
+// credentialTypesMatchOffer performs format specific validation.
+func credentialTypesMatchOffer(reader jsonld.Reader, credential vc.VerifiableCredential, offeredCredential map[string]interface{}) error {
+	switch offeredCredential["format"] {
+	case oidc4vci.VerifiableCredentialJSONLDFormat:
+		// In json-LD format the types need to be compared in expanded format
+		document, err := reader.Read(offeredCredential["credential_definition"])
+		if err != nil {
+			return fmt.Errorf("invalid credential_definition in offer: %w", err)
+		}
+		// TODO: can credentialDefinition contain invalid values that makes this panic?
+		expectedTypes := document.ValueAt(jsonld.NewPath("@type"))
+
+		document, err = reader.Read(credential)
+		if err != nil {
+			return fmt.Errorf("invalid credential: %w", err)
+		}
+		receivedTypes := document.ValueAt(jsonld.NewPath("@type"))
+
+		if !equal(expectedTypes, receivedTypes) {
```

Comment:
I think

`VerifiableCredential, NutsAuthorizationCredential` is equal to `NutsAuthorizationCredential, VerifiableCredential`.

Not very probable that it would differ, though, so probably OK (maybe add a note)

Reply:
I was not sure about this. According to the JSON spec an array is ordered, so I do not expect them to change from offer to issuance. The main question I had was how is this handled when fields are defined in 2 context, is the one that is applied selected based on the order of contexts, types (meaning that order must be preserved), or something else?

Reply:
Order is irrelevant, but is preserved. So probably OK like this (super-edge-case I'd say)